floss/main.py

```diff
@@ -955,13 +988,13 @@ def load_shellcode_workspace(sample_file_path, save_workspace, shellcode_ep_in,
                                            save_workspace, sample_file_path)
 
 
-def load_vw(sample_file_path, save_workspace, verbose, is_shellcode, shellcode_entry_point, shellcode_base):
+def load_vw(sample_file_path, complementary_idb_file_path, save_workspace, verbose, is_shellcode, shellcode_entry_point, shellcode_base):
```

Comment:
my ultimate goal is to read the vw directly from an idb, rather  than from the complementary idb. we should detect the file type here and dispatch to an appropriate loader.

Reply:
I agree, that would be a much better option.
I'll add it when https://github.com/williballenthin/python-idb has the required support.

Reply:
Also we would have to ensure that we could extract the 'original' binary from the idb to ensure the normal strings function would still work. I don't think the MZ and PE header are fully preserved but I might be wrong here.